java/src/main/java/io/cucumber/cucumberexpressions/ExpressionFactory.java

```diff
@@ -29,14 +27,25 @@ public ExpressionFactory(ParameterTypeRegistry parameterTypeRegistry) {
     }
 
     public Expression createExpression(String expressionString) {
-        if (BEGIN_ANCHOR.matcher(expressionString).find() || END_ANCHOR.matcher(expressionString).find()) {
-            return createRegularExpressionWithAnchors(expressionString);
+        int length = expressionString.length();
+        int lastCharIndex = 0;
+        char firstChar = 0;
+        char lastChar = 0;
+        if (length > 0) {
```

Comment:
Could we simplify this by handling the empty expression separately? This should remove the need to keep track of the `lastCharIndex` variable and would allow the `firstChar` and `lastChar` to be set only once.

 E.g:

```
if (expressionString.isEmpty()) {
  return new CucumberExpression(expressionString, this.parameterTypeRegistry);
}

Reply:
Thanks, I'll apply the suggestion but the `lastCharIndex` must be kept because it is used twice.

Reply:
I think the only usage would be this:

```
firstChar = expressionString.charAt(0);
lastChar = expressionString.charAt(expressionString.length() - 1);`
```

What am I missing?

Reply:
`Pattern.compile(expressionString.substring(1, lastCharIndex))` on line 44

Reply:
Ah. Right.